apps/rewards/app/components/App/App.js

```diff
@@ -41,6 +48,27 @@ class App extends React.Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if(nextProps.balances) {
+      this.updateConvertedRates(nextProps)
+    }
+  }
+
+  updateConvertedRates = throttle(async ({ balances }) => {
```

Comment:
Are you throttling here to rate limit the api calls?

Reply:
Yeah exactly. Basically, we don't want a new call going out every time the app state updates